From 8d0826759666dd0c207790b4ce8ac548bec760b8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20La=C3=9F?= <bevan@bi-co.net>
Date: Thu, 13 May 2021 13:22:57 +0200
Subject: [PATCH 3/4] padHandler: correct valid range for last

3a2addd introduced a check for last in padHandler::handle_variablePAD.
However, j only runs until xpadLength-1, so last being xpadLength-1 is
totally fine.
---
 src/backend/data/pad-handler.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/backend/data/pad-handler.cpp b/src/backend/data/pad-handler.cpp
index 570bb2e..c694fea 100755
--- a/src/backend/data/pad-handler.cpp
+++ b/src/backend/data/pad-handler.cpp
@@ -194,8 +194,8 @@ std::vector<uint8_t> data;		// for the local addition
 	if (CI_flag == 0) {
 	   if (mscGroupElement && (xpadLength > 0)) {
 
-	      if (last < xpadLength) {
-	         fprintf(stderr, "handle_variablePAD: last < xpadLength\n");
+	      if (last < xpadLength - 1) {
+	         fprintf(stderr, "handle_variablePAD: last < xpadLength - 1\n");
 	         return;
 	      }
 
-- 
2.31.1

